packages/react-dom/src/events/__tests__/SyntheticClipboardEvent-test.js

```diff
@@ -9,24 +9,39 @@
 
 'use strict';
 
-var SyntheticClipboardEvent;
+var jsdom;
+var React;
+var ReactDOM;
 
 describe('SyntheticClipboardEvent', () => {
-  var createEvent;
+  var document;
+  var simulateEvent;
 
   beforeEach(() => {
-    // TODO: can we express this test with only public API?
-    SyntheticClipboardEvent = require('SyntheticClipboardEvent');
-    createEvent = function(nativeEvent) {
-      var target = require('getEventTarget')(nativeEvent);
-      return SyntheticClipboardEvent.getPooled({}, '', nativeEvent, target);
+    jsdom = require('jsdom').jsdom;
+    React = require('react');
+    ReactDOM = require('react-dom');
+
+    document = jsdom('');
+
+    simulateEvent = (element, type, data) => {
+      var event = document.createEvent('Event');
+      event.initEvent(type, true, true);
+
+      if (data) {
+        event.clipboardData = data;
+      }
+
+      element.dispatchEvent(event);
```

Comment:
@gaearon Does it make any huge difference on our tests using the **Event** constructor rather than the old-fashioned **initEvent** way when we want to create events?

Actually, how is Fiber dealing with these two approaches today?

Reply:
I don't know if there's any difference. React just subscribes with `addEventListener`. If that works with both approaches in jsdom, let's use whichever is simpler.

Reply:
That was my first attempt, but I keep getting this error:
`TypeError: Argument to dispatchEvent must be an Event`

Reply:
@SadPandaBear Not sure if it's possible to use the `Event` constructor in this case, since I have to mock the `clipboardData`. Any ideas?

Reply:
I was thinking on doing something like:

```js
      var event = new ClipboardEvent(type, { bubbles: true });
      if(data) event.clipboardData = data;
      element.dispatchEvent(event);
```

But unfortunately `ClipboardEvent` is not supported. 😢